src/movies.js

```diff
@@ -1,22 +1,107 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(array) {
+  const allDirectors = array.map(function(movie){
+    return movie.director
+  })
+  return allDirectors;
+} 
+
+// allDirctors.filter(function(director, i){
+//   return allDirectors.indexOf(diector) == i
+// })
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+
+function howManyMovies(array) {
+  const filtered = array.filter(function(movie) { 
+
+     if (movie.director == 'Steven Spielberg' &&
+	     movie.genre.includes('Drama')) {
+	      return true
+     }
+ 
+    })
+    return filtered.length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(array) {
+  allScores = array.map(function(movie){
+    return movie.score
+  })
+  const totalScores = allScores.reduce(function(acc,curr){
+    return  acc+(curr||0)
+    
+  },0)
+  let Average
+  if (allScores.length == 0){
+    Average =0
```

Comment:
As of the naming convention, we should only name classes and React components with a capital letter.